### ISSUE TITLE
warning: failed to automatically apply fixes suggested by rustc to crate

### ISSUE THREAD

**Issue by rry-web** [rust-lang/rust-clippy#16426]
### Summary

Hello, this is my first time reporting an issue. Just following instructions that I got from clippy, if there's any additional information you want me to add that I'm missing I'd be happy to include it! I have the full logs attached below:

[clippylog.txt](https://github.com/user-attachments/files/24734291/clippylog.txt)

admin@Mac rynx_browser % cargo clippy --fix --allow-dirty
    Checking rynx_browser v0.1.0 (/Users/admin/rustprojects/rynx_browser)
warning: failed to automatically apply fixes suggested by rustc to crate `rynx_browser`

after fixes were automatically applied the compiler reported errors within these files:

  * src/event_handler.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0107]: function takes 0 generic arguments but 1 generic argument was supplied
  --> src/event_handler.rs:42:30
   |
42 |         InputMode::Normal => handle_normal_mode::<B>(app, key, terminal_width, terminal_height),
   |                              ^^^^^^^^^^^^^^^^^^----- help: remove the unnecessary generics
   |                              |
   |                              expected 0 generic arguments
   |
note: function defined here, with 0 generic parameters
  --> src/event_handler.rs:49:4
   |
49 | fn handle_normal_mode(
   |    ^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0107`.
Original diagnostics will follow.

### Reproducer

Code:

```rust
<code>
```

Current output:

Desired output:

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08) (Homebrew)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: aarch64-apple-darwin
release: 1.92.0
LLVM version: 21.1.7
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by samueltardieu** [comment]
Thanks for your report.

> if there's any additional information you want me to add that I'm missing I'd be happy to include it!

We would need the original code that triggered this bug if it can be shared, or a minimal reproducer if it cannot.

**Comment by rry-web** [comment]
[event_handler.txt](https://github.com/user-attachments/files/24748695/event_handler.txt)
Here's my event_handler.rs code, put in a text file so it can be uploaded here.

[copy.zip](https://github.com/user-attachments/files/24749214/copy.zip)
Here's the zipped folder of my full source code at the time I found the error in.
